src/Elcodi/Admin/PageBundle/Form/EventListener/PermanentPageSubscriber.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/*
+ * This file is part of the Elcodi package.
+ *
+ * Copyright (c) 2014 Elcodi.com
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * Feel free to edit as you please, and have fun.
+ *
+ * @author Marc Morera <yuhu@mmoreram.com>
+ * @author Aldo Chiecchia <zimage@tiscali.it>
+ */
+
+namespace Elcodi\Admin\PageBundle\Form\EventListener;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
+
+/**
+ * Class PermanentPageSubscriber checks if the page that's being handled by the form is persistent to modify the form
+ * fields removing the options that are not present for persistent pages.
+ */
+class PermanentPageSubscriber implements EventSubscriberInterface
+{
+    /**
+     * Gets the events which the class is subscribed to.
+     *
+     * @return array The array of events and methods to call.
+     */
+    public static function getSubscribedEvents()
+    {
+        return [FormEvents::PRE_SET_DATA => 'preSetData'];
+    }
+
+    /**
+     * This method is called during the Form::setData() call and removes the options that are not allowed for persistent
+     * pages.
+     *
+     * @param FormEvent $event The form event launched.
+     */
+    public function preSetData(FormEvent $event)
+    {
+        $page = $event->getData();
+        $form = $event->getForm();
+
+        if (!empty($page) && $page->isPersistent()) {
```

Comment:
:+1: for the `empty()`. Think about what happens if`$data` is equal to `"0"`.

From the [PHP Manual](http://php.net/manual/en/function.empty.php), `empty()++ matches the following:

```
"" (an empty string)
0 (0 as an integer)
0.0 (0 as a float)
"0" (0 as a string)
NULL
FALSE
array() (an empty array)
$var; (a variable declared, but without a value)
```